sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -374,6 +374,52 @@ case class ShowDatabasesCommand(databasePattern: Option[String]) extends Runnabl
 }
 
 /**
+ * A command for users to list the properties for a table.
+ * If propertyKey is specified, the value for the propertyKey
+ * is returned. If propertyKey is not specified, all the keys
+ * and their corresponding values are returned.
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   SHOW TBLPROPERTIES table_name[('propertyKey')];
+ * }}}
+ */
+case class ShowTablePropertiesCommand(
+    table: TableIdentifier,
+    propertyKey: Option[String]) extends RunnableCommand {
+
+  override val output: Seq[Attribute] = {
+    val withKeySchema: Seq[Attribute] = {
+      AttributeReference("value", StringType, nullable = false)() :: Nil
+    }
+    val noKeySchema: Seq[Attribute] = {
+      AttributeReference("key", StringType, nullable = false)() ::
+        AttributeReference("value", StringType, nullable = false)() :: Nil
+    }
+    propertyKey.map(p => withKeySchema).getOrElse(noKeySchema)
+  }
+
+  override def run(sqlContext: SQLContext): Seq[Row] = {
+    val catalog = sqlContext.sessionState.catalog
+
+    if (catalog.isTemporaryTable(table)) {
+      throw new AnalysisException("This operation is unsupported for temporary tables")
+    }
+    val catalogTable = sqlContext.sessionState.catalog.getTable(table)
+
+    propertyKey match {
+      case Some(p) =>
+        val errorStr = s"Table ${catalogTable.qualifiedName} does not have property: $p"
+        val propValue = catalogTable
+          .properties
+          .getOrElse(p, throw new AnalysisException(errorStr))
```

Comment:
Yeah.. Did think about it. In fact hive does not throw an error in this case. Do you prefer the resultant column to carry the exception string instead ?